API usage.md

```diff
@@ -0,0 +1,256 @@
+
+```js
+var y = 'y';
+x = x * 2;
+```
+
+### Compound
+
+`[left.name="x"][right.value=1]` - where the left side is a node name `x` and the right has the value `1`
+
+Valid: `x = 1`
+
+`[left.name="x"]:matches(*)` any type of node where the left side is named `x`
+
+### Descendant
+
+`Program IfStatement` - Any program with an `if` statement
+
+`Identifier[name=x]` - identified named `x` such as `const x`. Identifiers are used to name variables and functions and to provide labels for certain loops...
+
+`Identifier [name=x]` - any identifier with named `x`, such as `const x`
+
+`BinaryExpression [name=x]` a binary expression where one side is named `x`, such as `x !== 3`
+
+`AssignmentExpression [name=x]` an assignment where one side is named `x`, such as `x = 2` or `y = x`
+
+### Fields
+
+You can also query on the [Mozilla Parser API](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey/Parser_API) fields directly
+
+Example:
+
+```js
+interface IfStatement <: Statement {
+    type: "IfStatement";
+    test: Expression;
+    consequent: Statement;
+    alternate: Statement | null;
+}
+```
+
+`.test` - node (object) that has a `test` field set
+`.declarations.init` - node that has `.declarations.init` set
+
+`init` means initialised, such as `var x = 1`, where `x` is initialised to value `1`
+
+```js
+interface VariableDeclaration <: Declaration {
+    type: "VariableDeclaration";
+    declarations: [ VariableDeclarator ];
+    kind: "var" | "let" | "const";
+}
+
+interface VariableDeclarator <: Node {
+    type: "VariableDeclarator";
+    id: Pattern;
+    init: Expression | null;
+}
+```
+
+### :has
+
+`ExpressionStatement:has([name="foo"][type="Identifier"])`
+
+Valid: `const foo = 2`
+
+### Matches
+
+`,` means OR (ie. any of)
+
+- `:matches(IfStatement)`
+- `:matches(BinaryExpression, MemberExpression)`
+- `:matches([name="foo"], ReturnStatement)`
+- `:matches(AssignmentExpression, BinaryExpression)`
+- `AssignmentExpression, BinaryExpression, NonExistant`
+
+### Not
+
+- `:not(Literal)` not a literal
+- `:not([name="x"])` not a node named `x`
+- `:not(*)` - not anything!
+- `:not(Identifier, IfStatement)`
+- `:not([value=1])` not a node set to value of `1`
+
+### Pseudo child
+
+`:first-child` - first child node
+`:last-child` - last child node
+`:nth-child(2)` - 2nd child
+`:nth-last-child(2)` - 2nd last child
+
+### Subject
+
+`!IfStatement Identifier` - any not an If statement with an Identifier under, such as `const x = 3` but not `if (x == 2)`
+
+`!* > [name="foo"]` all nodes but those where the immediate child is a node namded `foo` 
+
+More examples:
+
+- `![test] [name="y"]`
+- `![generator=type(boolean)] > BlockStatement`
+- `![operator=/=+/] > [name="x"]`
+- `!:matches(*) > [name="foo"]`
+- `!:not(BlockStatement) > [name="foo"]`
+- `![left.name="x"][right.value=1]`
+- `* !AssignmentExpression`
```

Comment:
Please do whatever it takes to add sufficient docs for how to use the API. I tried my best using the tests cases for inspiration.